### PR TITLE
Support assigning `Results` to `List` properties via KVC

### LINKED ISSUE #2634: Assign Results to List using subscript
```swift
class Conversation: Object {
    let messages = List<Message>()
}

conversation["messages"] = messages
```

This works fine if `messages` is a `RLMArray`, `RLMListBase`, or `NSArray` but does not work if it is a `Results`. It throws an exception in `RLMDynamicValidatedSet`.

As far as I can tell, `RLMDynamicSet` only requires a collection that conforms to `NSFastEnumeration` which `ResultsBase` and `RLMResults` do.

It whole be helpful if you could assign using `Results` or `RLMResults` too.

My current workaround is to create an `Array` from the `Results` before assigning

```swift
let messages = //some query
conversation["messages"] = Array(messages)
```

### PR DESCRIPTION
**Pull request by jpsim**
Fixes #2634.

Ideally we'd support this via RealmCollectionType as well, but I couldn't really get that to work. /cc @tgoyne @bdash

### CODE REVIEW

**Comment by tgoyne** [comment]
Would be good to have an obj-c test as well since the change looks like it should cover that, but otherwise :+1:.

**Comment by jpsim** [comment]
objc test added.

**Comment by jpsim** [comment]
Thoughts on this @bdash?

File: RealmSwift-swift1.2/Tests/RealmCollectionTypeTests.swift [hunk 1]
```diff
@@ -302,9 +306,35 @@ class RealmCollectionTypeTests: TestCase {
...
+class ResultsTests: RealmCollectionTypeTests {
+    override class func defaultTestSuite() -> XCTestSuite {
+        // Don't run tests for the base class
+        if isEqual(ResultsTests) {
+            return XCTestSuite(name: "empty")
+        }
+        return super.defaultTestSuite()
+    }
+
+    func base() -> Results<SwiftStringObject> {
```
Comment (bdash) [inline RealmSwift-swift1.2/Tests/RealmCollectionTypeTests.swift]: I'm not clear what the name "base" represents here. Base of what?

**Comment by bdash** [comment]
:+1:

The Swift test changes here are another place where more detail in the commit message would have been appreciated.

**Comment by jpsim** [comment]
Good points on providing more details, I'll make an effort to improve that in the future.

**Comment by jpsim** [comment]
CI failure unrelated (#2119).

**Pull request merged**